pkg/controllers/job/job_controller_actions.go

```diff
@@ -465,7 +472,15 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 		RetryCount:          job.Status.RetryCount,
 	}
 
-	if updateStatus != nil && updateStatus(&job.Status) {
+	if updateStatus != nil {
+		updateStatus(&newStatus)
+	}
+
+	if reflect.DeepEqual(job.Status, newStatus) {
+		klog.V(3).Infof("Job <%s/%s> has not updated for no changing", job.Namespace, job.Name)
+		return nil
+	} else {
```

Comment:
Let's remove the else.

Reply:
done

Reply:
I mean code in else can be out of if else case. The `else {` is not nessasary any more.

Reply:
ok, I have done